Backpack/BottomSheet/Classes/PersistentBottomSheetLayout.swift

```diff
@@ -0,0 +1,44 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2020 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FloatingPanel
+
+public final class PersistentBottomSheetLayout: FloatingPanelLayout {
+    public var initialPosition: FloatingPanelPosition {
+        return .half
+    }
+
+    public var supportedPositions: Set<FloatingPanelPosition> {
+        return [.half, .full, .tip]
+    }
+
+    public func insetFor(position: FloatingPanelPosition) -> CGFloat? {
+        switch position {
+        case .half:
+            return 380
```

Comment:
How does this work on e.g. iPhone Se first gen? It's only 586pts tall. Maybe this logic needs to be device aware. Similarily how does this all work on iPad?

Reply:
My reasoning for this fixed height was based on what was previously done for the `Modal` presentation style. We had this constant `bottomSheetHeightInHalfPosition` set to `386.0`

Maybe we could let either FloatingPanel decide which will be the "half" height position or create a feature request where BottomSheet is aware of his content and set an appropriate half position. Let me know which option you consider is the best approach.

Reply:
I guess if you have just ported over existing behaviour I can't very well hold it over you anyway 👍🏼 I'll see if we can fix this later

Reply:
Shouldn't this be `Constants.bottomSheetHeightInHalfPosition` then? Which is 386 and not 380

Reply:
Yes. I think I did this just for a particular adjustment on the content in the feature that I was working on. I'm going to be consistent here.